src/main/java/com/linkedin/kmf/consumer/NewConsumerKMBaseConsumerFactoryImpl.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2020 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.linkedin.kmf.consumer;
+
+import com.linkedin.kmf.services.configs.CommonServiceConfig;
+import com.linkedin.kmf.services.configs.ConsumeServiceConfig;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class NewConsumerKMBaseConsumerFactoryImpl implements KMBaseConsumerFactory {
```

Comment:
Rather than use an altogether new factory impl, you could do this in the existing one using reflection - i.e., if `getConstructor(Map.class)` does not exist, then fall back to `getConstructor(String.class, Properties.class, AdminClient.class)`

That said, I don't really understand how this is intended to be used. i.e., what uses `ConsumerFactoryImpl` and what uses `NewConsumerKMBaseConsumerFactoryImpl` ?

Also, why does `ConsumerFactory` contain methods such as `latencySlaMs()`, etc.? Seems a bit weird to me.

I haven't been following all the KMF patches lately, so sorry about my ignorance on the current state.

Reply:
Within ConsumeService, we reference these instance variables from `ConsumerFactory.`
```
_latencySlaMs = consumerFactory.latencySlaMs(); 
_adminClient = consumerFactory.adminClient();
 ....
consumerFactory.latencyPercentileMaxMs(),
consumerFactory.latencyPercentileGranularityMs());
_commitLatencyMetrics = new CommitLatencyMetrics(metrics, tags, consumerFactory.latencyPercentileMaxMs(),
```

Reply:
I see - it is slightly weird in that a `factory<t>` should ideally only deal with dispensing instances of `t`. Here it appears to be used to stash away some additional config information that is used elsewhere. It's NBD though, and orthogonal to what you are trying to do in this patch, so I'll just ignore that :)